packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -245,9 +245,13 @@ export class AngularWebpackPlugin {
         .filter((sourceFile) => !internalFiles?.has(sourceFile));
 
       // Ensure all program files are considered part of the compilation and will be watched
-      allProgramFiles.forEach((sourceFile) =>
-        compilation.compilationDependencies.add(sourceFile.fileName),
-      );
+      allProgramFiles.forEach((sourceFile) => {
+        if (isWebpackFiveOrHigher()) {
```

Comment:
Wouldn't a feature check be better?
```suggestion
        if (compilation.fileDependencies) {
```

Reply:
It can actually be present in both cases in certain situations.  In webpack 4 it is added as a property during the compilation process based on the contents of `compilationDependencies`.  Out of an abundance of caution, I went with this method to avoid the potential for hard to diagnose issues in the future. And once Webpack 5 is the default, it can be removed completely.  `isWebpackFiveOrHigher` is also cached so it reduces to a boolean check after the first use.